File: Data-Structures-February-2016/09. Data-Structure-Efficiency/[HW]/01. Students-And-Courses/Person.cs
namespace StudentsAndCourses
{
    using System;

    public class Person : IComparable<Person>
    {
        public Person(string firstName, string lastName)
        {
            this.FirstName = firstName;
            this.LastName = lastName;
        }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public int CompareTo(Person other)
        {
            int compareIndex = this.LastName.CompareTo(other.LastName);
            if (compareIndex == 0)
            {
                compareIndex = this.FirstName.CompareTo(other.FirstName);
            }

            return compareIndex;
        }

        public override string ToString()
        {
            return this.FirstName + " " + this.LastName;
        } 
    }
}

File: 1. Fundamental-Level/Java/Advanced-OPP-July-2016/03. Iterators&Comparators/src/problem5_ComparingObjects/Person.java
package problem5_ComparingObjects;

public class Person implements Comparable<Person> {

    private String name;
    private int age;
    private String town;

    public Person(String name, int age, String town) {
        this.name = name;
        this.age = age;
        this.town = town;
    }

    @Override
    public int compareTo(Person other) {
        int compareIndex = this.name.compareTo(other.name);
        if (compareIndex == 0) {
            compareIndex = Integer.compare(this.age, other.age);
            if (compareIndex == 0) {
                compareIndex = this.town.compareTo(other.town);
            }
        }

        return compareIndex;
    }

    @Override
    public String toString() {
        return String.format("%s %d %s",
                this.name, this.age, this.town);
    }
}
